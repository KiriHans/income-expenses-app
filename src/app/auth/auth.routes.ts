import { Routes } from '@angular/router';
import { LoginComponent } from './components/login/login.component';
import { RegisterComponent } from './components/register/register.component';
import { adminGuard } from '../core/guards/admin.guard';

export default [
  {
    path: 'auth',
    canActivate: [adminGuard],
    children: [
      {
        path: 'login',
        component: LoginComponent,
      },
      {
        path: 'register',
        component: RegisterComponent,
      },
    ],
  },
] satisfies Routes;
