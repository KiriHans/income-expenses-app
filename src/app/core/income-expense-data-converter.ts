import {
  FirestoreDataConverter,
  QueryDocumentSnapshot,
  SnapshotOptions,
} from '@angular/fire/firestore';
import { IncomeExpense } from './models/income-expenses.model';

export const incomeExpenseConverter: FirestoreDataConverter<IncomeExpense> = {
  toFirestore: (incomeExpense: IncomeExpense) => {
    const { description, amount, type, uid } = incomeExpense;
    return { description, amount, type, uid };
  },
  fromFirestore: (
    snapshot: QueryDocumentSnapshot<IncomeExpense>,
    options: SnapshotOptions | undefined
  ) => {
    const data = snapshot.data(options);
    return new IncomeExpense(data.description, data.amount, data.type, data.uid);
  },
};
