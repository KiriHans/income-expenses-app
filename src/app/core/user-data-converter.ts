import {
  FirestoreDataConverter,
  QueryDocumentSnapshot,
  SnapshotOptions,
} from '@angular/fire/firestore';
import { UserApp } from './models/users.model';

export const userConverter: FirestoreDataConverter<UserApp> = {
  toFirestore: (user: UserApp) => {
    const { displayName, email } = user;
    return { displayName, email };
  },
  fromFirestore: (
    snapshot: QueryDocumentSnapshot<UserApp>,
    options: SnapshotOptions | undefined
  ) => {
    const data = snapshot.data(options);
    return new UserApp(data.id, data.displayName, data.email);
  },
};
