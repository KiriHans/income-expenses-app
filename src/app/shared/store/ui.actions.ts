import { createActionGroup, emptyProps } from '@ngrx/store';

export const UiActions = createActionGroup({
  source: 'UI',
  events: {
    'Load UI': emptyProps(),
    'Stop Load UI': emptyProps(),
  },
});
