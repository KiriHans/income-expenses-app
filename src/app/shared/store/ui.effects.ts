import { Injectable } from '@angular/core';
import { Actions, createEffect, ofType } from '@ngrx/effects';
import { map, concatMap } from 'rxjs/operators';
import { EMPTY } from 'rxjs';
import { UiActions } from './ui.actions';

@Injectable()
export class UiEffects {
  loadUis$ = createEffect(() => {
    return this.actions$.pipe(
      ofType(UiActions.loadUI),
      concatMap(() =>
        /** An EMPTY observable only emits completion. Replace with your own observable API request */
        EMPTY.pipe(map((data) => UiActions.loadUI()))
      )
    );
  });

  constructor(private actions$: Actions) {}
}
