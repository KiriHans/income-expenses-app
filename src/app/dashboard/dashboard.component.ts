import { Component, OnInit, Signal } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterOutlet } from '@angular/router';
import { FooterComponent } from '../shared/components/footer/footer.component';
import { NavbarComponent } from '../shared/components/navbar/navbar.component';
import { SidebarComponent } from '../shared/components/sidebar/sidebar.component';
import { UserApp } from '../core/models/users.model';
import { Store } from '@ngrx/store';
import { authFeature } from '../store/reducers/auth.reducer';

@Component({
  selector: 'app-dashboard',
  standalone: true,
  imports: [CommonModule, RouterOutlet, FooterComponent, NavbarComponent, SidebarComponent],
  templateUrl: './dashboard.component.html',
  styles: [],
})
export class DashboardComponent implements OnInit {
  user$!: Signal<UserApp>;

  constructor(private store: Store) {}

  ngOnInit(): void {
    this.user$ = this.store.selectSignal(authFeature.selectUserState);
  }
}
