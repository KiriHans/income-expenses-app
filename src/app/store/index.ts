import { isDevMode } from '@angular/core';
import { ActionReducerMap, MetaReducer } from '@ngrx/store';
import * as ui from '../shared/store/ui.reducer';
import * as auth from './reducers/auth.reducer';
import * as incomeExpense from './reducers/income-expense.reducer';

export const stateFeatureKey = 'state';

export interface AppState {
  [ui.uiFeatureKey]: ui.UIState;
  [auth.authFeatureKey]: auth.State;
  [incomeExpense.incomeExpenseFeatureKey]: incomeExpense.State;
}

export const reducers: ActionReducerMap<AppState> = {
  [ui.uiFeatureKey]: ui.uiFeature.reducer,
  [auth.authFeatureKey]: auth.userReducer,
  [incomeExpense.incomeExpenseFeatureKey]: incomeExpense.reducer,
};

export const metaReducers: MetaReducer<AppState>[] = isDevMode() ? [] : [];
