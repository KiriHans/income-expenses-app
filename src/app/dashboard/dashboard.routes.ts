import { Routes } from '@angular/router';
import { DashboardComponent } from './dashboard.component';
import INCOME_EXPENSES_ROUTES from '../income-expense/income-expense.routes';
import { authGuard } from '../core/guards/auth.guard';

export default [
  {
    path: '',
    component: DashboardComponent,
    children: INCOME_EXPENSES_ROUTES,
  },
] satisfies Routes;
