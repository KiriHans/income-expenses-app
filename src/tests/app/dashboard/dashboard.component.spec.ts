import { ComponentFixture, TestBed } from '@angular/core/testing';

import { RouterTestingModule } from '@angular/router/testing';

import * as Auth from 'src/app/store/reducers/auth.reducer';

import { provideMockStore, MockStore } from '@ngrx/store/testing';
import { DashboardComponent } from 'src/app/dashboard/dashboard.component';

describe('DashboardComponent', () => {
  let component: DashboardComponent;
  let fixture: ComponentFixture<DashboardComponent>;

  let store: MockStore;

  const initialState = {
    user: {
      id: '123',
      displayName: 'test',
      email: 'test@gm.com',
      loading: false,
    },
  };

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [DashboardComponent, RouterTestingModule],
      providers: [provideMockStore({ initialState })],
    });
    fixture = TestBed.createComponent(DashboardComponent);
    component = fixture.componentInstance;

    store = TestBed.inject(MockStore);

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    console.log(component.user$());
  });
});
