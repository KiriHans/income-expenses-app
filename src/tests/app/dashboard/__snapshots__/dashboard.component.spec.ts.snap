// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <app-navbar>
    <div
      class="navbar"
    >
      test, test@gm.com
    </div>
  </app-navbar>
  <div
    class="page-body-wrapper"
  >
    <app-sidebar>
      <div
        class="sidebar"
      >
        test, test@gm.com
      </div>
    </app-sidebar>
    <div
      class="main-panel"
    >
      <div
        class="content-wrapper"
      >
        <router-outlet />
        <app-dumb>
          <div />
        </app-dumb>
        
      </div>
      <app-footer>
        <footer
          class="footer"
        >
          <div
            class="container-fluid clearfix"
          >
            <span
              class="text-muted d-block text-center text-sm-left d-sm-inline-block"
            >
               All rights reserved @ 2023 
              <a
                href="https://github.com/KiriHans"
                target="_blank"
              >
                Steven Rojas
              </a>
              . 
            </span>
            <span
              class="float-none float-sm-right d-block mt-1 mt-sm-0 text-center"
            >
               Created by hand with 
              <i
                class="fa fa-heart text-danger"
              />
            </span>
          </div>
        </footer>
      </app-footer>
    </div>
  </div>
</div>
`;
